google/cloud/pubsub/internal/streaming_subscription_batch_source_test.cc

```diff
@@ -493,22 +516,55 @@ TEST(StreamingSubscriptionBatchSourceTest, WriteErrorWaitsForRead) {
   uut->AckMessage("fake-001", 0);
   auto pending_write = fake_stream.WaitForAction();
 
+  shutdown->MarkAsShutdown("test", expected_status);
+  uut->Shutdown();
+
   pending_write.set_value(false);
   pending_read.set_value(false);
 
-  shutdown->MarkAsShutdown("test", expected_status);
+  EXPECT_EQ(expected_status, done.get());
+}
+
+TEST(StreamingSubscriptionBatchSourceTest, ShutdownWithPendingRead) {
+  auto subscription = pubsub::Subscription("test-project", "test-subscription");
+  std::string const client_id = "fake-client-id";
+  AutomaticallyCreatedBackgroundThreads background;
+  auto mock = std::make_shared<pubsub_testing::MockSubscriberStub>();
+
+  auto const expected_status = Status{StatusCode::kOk, "fine"};
```

Comment:
Q. Should we ever be constructing an OK `Status` that contains a message?

Reply:
No, fixed.